docs/user/openstack/etcd-ephemeral-disk.md

```diff
@@ -0,0 +1,203 @@
+# Moving etcd to an ephemeral local disk
+
+You can move etcd from a root volume (Cinder) to a dedicated ephemeral local disk to prevent or resolve performance issues.
+
+## Prerequisites
+
+* An OpenStack cloud with Cinder being functional and enough available storage (at least 75 GB) to accommodate 3 Root Volumes for the OpenShift control plane.
+* An OpenStack cloud whith Nova not using Ceph as a backend for ephemeral disks.
+* OpenShift will be deployed with IPI for now; UPI is not yet documented but technically possible.
+* The MachineConfigPool must match metadata.labels[machineconfiguration.openshift.io/role]. This applies to a controller, worker, or a custom pool.
+* The control-plane machine’s auxiliary storage device, such as /dev/vdb, must match the vdb. Change this reference in all places in the file.
+
+## Procedure
+
+* Create a Nova flavor for the Control Plane which allows 10 GiB of Ephemeral Disk:
+
+```bash
+openstack flavor create --ram 16384 --disk 25 --ephemeral 10 --vcpus 4 m1.xlarge
+```
+
+* If you are deploying a new cluster, run openshift-install with the following parameters to create the manifests:
+
+```bash
+openshift-install create manifests --dir=install_dir
+```
+
+* If you are deploying a new cluster, edit the ControlPlaneMachineSet (CPMS) in `install-dir/openshift/99_openshift-machine-api_master-control-plane-machine-set.yaml` to add the additional block ephemeral device that will be used by etcd:
+
+```yaml
+[...]
+providerSpec:
+  additionalBlockDevices:
+  - name: etcd
+    sizeGiB: 10
+    storage:
+      type: Local
+[...]
```

Comment:
What do you think about adding another example of `additionalBlockDevices` with volume type given this is a new API? Perhaps mention it in the customization docs to not mix with etcd things?

Reply:
I don't want to advertise that simply because:

1. Our CI doesn't and won't test it in the short term.
2. putting etcd on a volume is not recommended at the moment.